horovod/spark/common/store.py

```diff
@@ -311,10 +315,35 @@ def sync_fn(self, run_id):
 
         def fn(local_run_path):
             print(f"Syncing dir {local_run_path} to dir {run_path}")
-            self.fs.put(local_run_path, run_path, recursive=True, overwrite=True)
+            if self.fs.exists(run_path):
+                local_run_path = os.path.join(local_run_path, self.get_wildcard())
+            self.copy(local_run_path, run_path, recursive=True, overwrite=True)
```

Comment:
Why this `copy()` function is not required in HDFS store?

Reply:
HDFS store's ```sync_fn()``` does not make use of ```put()```. While ```put()``` copies the source folder along with it's contents to the target folder when the target folder exists, this issue is eliminated in the HDFS store's ```sync_fn()``` because it copies from the source folder to the target folder only.